mp/src/game/shared/momentum/weapon/weapon_mom_rocketlauncher.cpp

```diff
@@ -36,11 +53,59 @@ void CMomentumRocketLauncher::Precache()
 {
     BaseClass::Precache();
 
+    m_iMomViewIndex = PrecacheModel(MOM_ROCKETLAUNCHER_MODEL);
+    m_iMomWorldIndex = PrecacheModel(MOM_ROCKETLAUNCHER_WMODEL);
+    m_iTFViewIndex = PrecacheModel(TF_ROCKETLAUNCHER_MODEL);
+    m_iTFWorldIndex = PrecacheModel(TF_ROCKETLAUNCHER_WMODEL);
+
 #ifndef CLIENT_DLL
     UTIL_PrecacheOther("momentum_rocket");
 #endif
 }
 
+bool CMomentumRocketLauncher::Deploy()
+{
+    CMomentumPlayer *pOwner = GetPlayerOwner();
+    int iModelToUse = 0;
+
+#ifdef CLIENT_DLL
+    static ConVarRef mom_rj_viewmodel("mom_rj_viewmodel");
+    iModelToUse = mom_rj_viewmodel.GetBool();
+#else
+    iModelToUse = mom_rj_viewmodel.GetBool();
+#endif
+    if (pOwner)
+    {
+        iModelToUse ? SetTFModel() : SetMomModel();
+    }
+
+    return BaseClass::Deploy();
+}
+
+const char *CMomentumRocketLauncher::GetViewModel(int) const
+{
+    return m_iViewModelIndex == m_iTFViewIndex ? TF_ROCKETLAUNCHER_MODEL : MOM_ROCKETLAUNCHER_MODEL;
+}
+
+const char *CMomentumRocketLauncher::GetWorldModel() const
+{
+    return m_iWorldModelIndex == m_iTFWorldIndex ? TF_ROCKETLAUNCHER_WMODEL : MOM_ROCKETLAUNCHER_WMODEL;
+}
+
+void CMomentumRocketLauncher::SetMomModel()
+{
+    m_iViewModelIndex = m_iMomViewIndex;
+    m_iWorldModelIndex = m_iMomWorldIndex;
+    SetModel(GetViewModel());
+}
+
+void CMomentumRocketLauncher::SetTFModel()
+{
+    m_iViewModelIndex = m_iTFViewIndex;
+    m_iWorldModelIndex = m_iTFWorldIndex;
+    SetModel(GetViewModel());
+}
```

Comment:
This and `SetMomModel` are functionally equivalent. Pull it out into a `SetModelType(bool bTF2Model)` void:
```
void CMomentumRocketLauncher::SetModelType(bool bTF2Model)
{
    m_iViewModelIndex = bTF2Model ? m_iTFViewIndex : m_iMomViewIndex;
    m_iWorldModelIndex = bTF2Model ? m_iTFWorldIndex : m_iMomWorldIndex;
    SetModel(GetViewModel());
}
```

Reply:
I get the following error for all four variables when building this:
`error C2446: ':': no conversion from 'CNetworkVarBase<int,CMomentumRocketLauncher::NetworkVar_m_iMomViewIndex>' to 'CNetworkVarBase<int,CMomentumRocketLauncher::NetworkVar_m_iTFViewIndex>'`

If I cast them to int like so it works fine:
`m_iViewModelIndex = bTF2Model ? (int) m_iTFViewIndex : (int) m_iMomViewIndex;
    m_iWorldModelIndex = bTF2Model ? (int) m_iTFWorldIndex : (int) m_iMomWorldIndex;`

Reply:
Add `.Get()` to the end of the vars instead of casting them.